pkg/chain/chain.go

```diff
@@ -30,135 +86,162 @@ type Handle interface {
 	// In case the block is not yet mined, an error should be returned.
 	BlockTimestamp(blockNumber *big.Int) (uint64, error)
 
-	BondedECDSAKeepFactory
-	BondedECDSAKeep
+	PublicKeyToOperatorID(publicKey *cecdsa.PublicKey) OperatorID
+
+	// BondedECDSAKeepManager returns a handle to the on-chain component used to
+	// interact with bonded ECDSA keeps.
+	BondedECDSAKeepManager() (BondedECDSAKeepManager, error)
 }
 
-// BondedECDSAKeepFactory is an interface that provides ability to interact with
-// BondedECDSAKeepFactory ethereum contracts.
-type BondedECDSAKeepFactory interface {
-	// RegisterAsMemberCandidate registers client as a candidate to be selected
-	// to a keep.
-	RegisterAsMemberCandidate(application common.Address) error
+// BondedECDSAKeepManager is an interface whose implementations allow for
+// managing, monitoring, and creating bonded ECDSA keeps as well as the
+// operator's eligibility for participation in such keeps.
+type BondedECDSAKeepManager interface {
+	// TBTCApplicationHandle returns a handle for interacting with the tBTC
+	// application associated with this BondedECDSAKeepManager. Returns nil with
+	// an error if no tBTC application exists for this manager.
+	TBTCApplicationHandle() (TBTCHandle, error)
 
 	// OnBondedECDSAKeepCreated installs a callback that is invoked when an
 	// on-chain notification of a new bonded ECDSA keep creation is seen.
 	OnBondedECDSAKeepCreated(
 		handler func(event *BondedECDSAKeepCreatedEvent),
 	) subscription.EventSubscription
 
+	// GetKeepCount returns number of keeps.
+	GetKeepCount() (*big.Int, error)
+
+	// GetKeepAtIndex returns a handle to the keep at the given index.
+	GetKeepAtIndex(keepIndex *big.Int) (BondedECDSAKeepHandle, error)
+	// GetKeepAtIndex returns a handle to the keep with the given identifier.
+	GetKeepWithID(keepID KeepID) (BondedECDSAKeepHandle, error)
+
+	// IsOperatorAuthorized checks if the factory has the authorization to
+	// operate on stake represented by the provided operator.
+	IsOperatorAuthorized(operator OperatorID) (bool, error)
+}
+
+type BondedECDSAKeepApplicationHandle interface {
```

Comment:
Where/How can I obtain an instance of this interface?

Reply:
TBTCApplicationHandle embeds it---I would expect the same to happen for any other application. So basically, off of the BondedECDSAKeepManager.